scoala-online/main
# Feat/#111 implement swgger UI

### This pull request solves issue:

Closes: #111 

### Which app is this issue related to?


### Types of changes


### Wiki Updates


### Testing checklist


#### Security Filters - **_For API Only_**

**_Please make sure to add the respective Security Filters before submitting a PR_**


#### Compatibility tests - **_For E-Learning React App Only_**


### What changes have been done within this issue?

I've added the necessary dependencies found under the  field.
I've created a config package where I added the config class for swagger `SpringFoxConfig.java`.
In `SecurityConfig.java` I've added a security layer for `/swagger-ui` endpoint.

### How should this be manually tested?

Start the application, then go to `http://localhost:5000/swagger-ui/`. You will have to be logged in with an administrator account, otherwise, a login form will appear that requires you to sign in. Afterward, you can go through the endpoint controllers and perform the listed HTTP requests by pressing Try it out -> Execute.

### Screenshots or example usage
